pkg/apis/core/validation/validation.go

```diff
@@ -352,6 +352,14 @@ func ValidateImmutableField(newVal, oldVal interface{}, fldPath *field.Path) fie
 	return apimachineryvalidation.ValidateImmutableField(newVal, oldVal, fldPath)
 }
 
+func ValidateImmutableLabel(newVal string, oldVal string, label string, fldPath *field.Path) field.ErrorList {
```

Comment:
Is this function really necessary? We have `ValidateImmutableField` in apimachineryvalidation? Could we use that somehow?

Reply:
It is not needed, I could call directly `ValidateImmutableField`, but I think it makes sense to have `ValidateImmutableLabel` to avoid repeating `fldPath.Child("labels", label)`. Note that just below we already have the `ValidateImmutableAnnotation` function.